lightning/src/routing/router.rs

```diff
@@ -4956,6 +5109,104 @@ mod tests {
 		}
 	}
 
+	#[test]
+	fn adds_and_limits_cltv_offset() {
+		let (secp_ctx, network_graph, _, _, logger) = build_graph();
+		let (_, our_id, _, nodes) = get_nodes(&secp_ctx);
+
+		let scorer = test_utils::TestScorer::with_penalty(0);
+
+		let payment_params = PaymentParameters::from_node_id(nodes[6]).with_route_hints(last_hops(&nodes));
+		let keys_manager = test_utils::TestKeysInterface::new(&[0u8; 32], Network::Testnet);
+		let random_seed_bytes = keys_manager.get_secure_random_bytes();
+		let route = get_route(&our_id, &payment_params, &network_graph.read_only(), None, 100, 42, Arc::clone(&logger), &scorer, &random_seed_bytes).unwrap();
+		let cltv_expiry_deltas_before = route.paths.iter().flat_map(|p| p.iter().map(|h| h.cltv_expiry_delta).collect::<Vec<u32>>()).collect::<Vec<u32>>();
```

Comment:
Will there only be on path in the route? If so, could we assert that and take `route.paths[0]` and avoid some of this code?

Reply:
Alright, simplified this in https://github.com/lightningdevkit/rust-lightning/pull/1286/commits/0b777cbd4ac95d1578264dab82826d725d2cd3d3.